src/lib.rs

```diff
@@ -423,23 +469,85 @@ impl NumToA for u8 {
     }
 }
 
+macro_rules! impl_numtoa_base_n_init_for {
+(
+    $type_name:ty,
+    $base:expr,
+    $core_function_name:ident,
+    $base_n_function_name:ident,
+    $needed_buffer_size:expr) => {
+
+        pub const fn $base_n_function_name(num: $type_name) -> AsciiNumber<$needed_buffer_size> {
+            let mut string = [0_u8; $needed_buffer_size];
+            let len = $core_function_name(num, 10, &mut string).len();
+            return AsciiNumber { string, start:$needed_buffer_size-len}
+        }
+
+    };
+}
+
+pub mod base10 {
+
+    use AsciiNumber;
+    use numtoa_u8;
+    use numtoa_u16;
+    use numtoa_u32;
+    use numtoa_u64;
+    use numtoa_u128;
+    use numtoa_i8;
+    use numtoa_i16;
+    use numtoa_i32;
+    use numtoa_i64;
+    use numtoa_i128;
+
+    impl_numtoa_base_n_init_for!(u8,10,numtoa_u8,u8,3); // 255
+    impl_numtoa_base_n_init_for!(u16,10,numtoa_u16,u16,5); // 65535
+    impl_numtoa_base_n_init_for!(u32,10,numtoa_u32,u32,10); // 4294967295
+    impl_numtoa_base_n_init_for!(u64,10,numtoa_u64,u64,20); // 18446744073709551615
+    impl_numtoa_base_n_init_for!(u128,10,numtoa_u128,u128,39); // 340282366920938463463374607431768211455
+    impl_numtoa_base_n_init_for!(i8,10,numtoa_i8,i8,4); // -128
+    impl_numtoa_base_n_init_for!(i16,10,numtoa_i16,i16,6); // -32768
+    impl_numtoa_base_n_init_for!(i32,10,numtoa_i32,i32,11); // -2147483648
+    impl_numtoa_base_n_init_for!(i64,10,numtoa_i64,i64,20); // -9223372036854775808
+    impl_numtoa_base_n_init_for!(i128,10,numtoa_i128,i128,40); // -170141183460469231731687303715884105728
+
+}
```

Comment:
moving forward, we can add one of these chunks for each base with the explicitly defined buffer sizes